src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineWindowsLauncher.java

```diff
@@ -0,0 +1,269 @@
+        }
+    }
+
+    protected void logInfo(ComputeEngineComputer computer, TaskListener listener, String message) {
+        log(Level.INFO, computer, listener, message);
+    }
+
+    protected void logWarning(ComputeEngineComputer computer, TaskListener listener, String message) {
+        log(Level.WARNING, computer, listener, message);
+    }
+
+    protected void launch(ComputeEngineComputer computer, TaskListener listener, Instance inst)
+            throws IOException, InterruptedException {
+        ComputeEngineInstance node = computer.getNode();
+        if (node == null) {
+            logWarning(computer, listener, "Could not get node from computer");
+            return;
+        }
+
+        final Connection bootstrapConn;
+        final Connection conn;
+        Connection cleanupConn = null; // java's code path analysis for final
+        // doesn't work that well.
+        boolean successful = false;
+        PrintStream logger = listener.getLogger();
+        logInfo(computer, listener, "Launching instance: " + node.getNodeName());
+        try {
+            boolean isBootstrapped = bootstrap(computer, listener);
+            if (isBootstrapped) {
+                // connect fresh as ROOT
+                logInfo(computer, listener, "connect fresh as root");
+                cleanupConn = connectToSsh(computer, listener);
+                if (!authenticateSSH(node.windowsConfig, cleanupConn, listener)) {
+                    logWarning(computer, listener, "Authentication failed");
+                    return; // failed to connect
+                }
+            } else {
+                logWarning(computer, listener, "bootstrapresult failed");
+                return;
+            }
+            conn = cleanupConn;
+
+            SCPClient scp = conn.createSCPClient();
+            String tmpDir = "C:\\";
+            logInfo(computer, listener, "Copying slave.jar to: " + tmpDir);
+            scp.put(Jenkins.getInstance().getJnlpJars("slave.jar").readFully(), "slave.jar", tmpDir);
+
+            // Confirm Java is installed
+            if (!testCommand(computer, conn, "java -fullversion", logger, listener)) {
+                logWarning(computer, listener, "Java is not installed.");
+            }
+
+            //TODO: allow jvmopt configuration
+            String launchString = "java -jar C:\\slave.jar";
+
+            logInfo(computer, listener, "Launching Jenkins agent via plugin SSH: " + launchString);
+            final Session sess = conn.openSession();
+            sess.execCommand(launchString);
+            computer.setChannel(sess.getStdout(), sess.getStdin(), logger, new Channel.Listener() {
+                @Override
+                public void onClosed(Channel channel, IOException cause) {
+                    sess.close();
+                    conn.close();
+                }
+            });
+        } catch (Exception e) {
+            logException(computer, listener, "Error getting exception", e);
```

Comment:
This error message could be more descriptive.

Reply:
Part of larger issue to have better exceptions.